pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -65,6 +65,7 @@ const (
 // AttachDetachController defines the operations supported by this controller.
 type AttachDetachController interface {
 	Run(stopCh <-chan struct{})
+	GetDesiredStateOfWorld() cache.DesiredStateOfWorld
```

Comment:
I'm not fully convinced that you need to add the function `GetDesiredStateOfWorld()` to the interface if all it does is return the desiredStateOfTheWorld held by the struct. It definitely has to be a function of `adc` but does it really have to be in the interface ?
I may be wrong and I'm not aware of what the best practice here is, but I feel there might be a better way of doing this.